### PR TITLE
Service role for stack installation and deletion

### PR DESCRIPTION
**Pull request by akarve**
Tested for stack creation and stack deletion with "all features on" using TF on a pure CF deployment

### CODE REVIEW

File: docs/cfn-service-role.yaml [hunk 1]
```diff
@@ -11,31 +11,51 @@ Resources:
...
-                  - apigateway:GetResources
                   - apigateway:PATCH
                   - apigateway:POST
                   - apigateway:PUT
+                  - apigateway:SetWebACL
                   - apigateway:UpdateRestApiPolicy
+                  - appsync:SetWebACL
+                  - apprunner:AssociateWebAcl
+                  - apprunner:DescribeWebAclForService
+                  - apprunner:DisassociateWebAcl
```
Comment (sir-sigurd) [inline docs/cfn-service-role.yaml:30]: where does this come from? 🤔

**Review by sir-sigurd — APPROVED** [review]
I still think it might make sense to just have `<service>:*` actions at this point

File: docs/cfn-service-role.yml [hunk 2]
```diff
@@ -91,19 +117,30 @@ Resources:
...
                   - es:DeleteElasticsearchDomain
                   - es:DescribeElasticsearchDomain
+                  - es:UpdateElasticsearchDomainConfig
                   - events:DeleteRule
                   - events:DescribeRule
+                  - events:ListTargetsByRule
                   - events:PutRule
                   - events:PutTargets
                   - events:RemoveTargets
+                  - firehose:CreateDeliveryStream
```
Comment (QuiltSimon) [inline docs/cfn-service-role.yml:145]: What are we using Kinesis Firehose for?

File: docs/cfn-service-role.yml [hunk 3]
```diff
@@ -135,42 +185,96 @@ Resources:
...
+                  - s3:GetBucketTagging
+                  - s3:GetBucketVersioning
+                  - s3:GetBucketWebsite
+                  - s3:GetEncryptionConfiguration
+                  - s3:GetIntelligentTieringConfiguration
+                  - s3:GetInventoryConfiguration
+                  - s3:GetLifecycleConfiguration
+                  - s3:GetMetricsConfiguration
                   - s3:GetObject
+                  - s3:GetReplicationConfiguration
```
Comment (QuiltSimon) [inline docs/cfn-service-role.yml]: Is this a currently used IAM policy? Or potentially future state?

**Review by QuiltSimon — APPROVED** [review]
ok

File: docs/cfn-service-role.yaml (hunk 1 above)
Comment (akarve) [reply docs/cfn-service-role.yaml:30]: That is apparently how WAF operations are dispatched. Specifically it was reported by IAM > Generate Policy.

File: docs/cfn-service-role.yml (hunk 2 above)
Comment (akarve) [reply docs/cfn-service-role.yml:145]: Canaries events AFAICT.

File: docs/cfn-service-role.yml (hunk 3 above)
Comment (akarve) [reply docs/cfn-service-role.yml]: We take * now (as the stack has). Believe it or not this is likely a current operation as reported by CloudTrail and Generate Policy.

File: docs/cfn-service-role.yaml (hunk 1 above)
Comment (sir-sigurd) [reply docs/cfn-service-role.yaml:30]: won't it work without this?

File: docs/cfn-service-role.yml (hunk 2 above)
Comment (sir-sigurd) [reply docs/cfn-service-role.yml:145]: audit trail

**Pull request merged**